crawler/dataParser.py

```diff
@@ -80,6 +91,27 @@ def runParser(self):
     for i in range(0, self.MAX_THREADS):
       threadPool[i].join()
 
+    self.saveLinkGraphs()
```

Comment:
Instead of writing to the file all at once, can you write to the file after a buffer of like 100 is filled up? This could be 1 buffer per thread, or 1 larger buffer total. You just gotta implement a write semaphore (youre in OS, you know what to do lmao)

Reply:
Nvm this is fine